### PR TITLE
IGNITE-12921 SqlViewExporterSpi moved to the internal package

### PR DESCRIPTION
**Pull request by nizhikov**
SqlViewExporterSpi moved to the internal package to eliminate it from public API.
Instances of this SPI will be created automatically in case the indexing module available during node startup.

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by nizhikov** [comment]
Hello, @agura can you please, take a look at my changes?

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java [hunk 1]
```diff
@@ -75,7 +82,26 @@
      * @param ctx Kernal context.
      */
     public GridSystemViewManager(GridKernalContext ctx) {
-        super(ctx, ctx.config().getSystemViewExporterSpi());
+        super(ctx, ((Supplier<SystemViewExporterSpi[]>)() -> {
```
Comment (agura) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java:85]: Could you please replace supplier by simple method call? There is no need to use lambda syntax here.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java [hunk 2]
```diff
@@ -75,7 +82,26 @@
...
+            SystemViewExporterSpi[] spi = ctx.config().getSystemViewExporterSpi();
+
+            if (!IgniteComponentType.INDEXING.inClassPath())
+                return spi;
+
+            SystemViewExporterSpi[] spiWithSql = new SystemViewExporterSpi[spi != null ? spi.length + 1 : 1];
```
Comment (agura) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java:91]: Nevertheless, what if `spi` array already contains `SqlViewExporterSpi`?

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java [hunk 3]
```diff
@@ -53,6 +57,9 @@
  */
 public class GridSystemViewManager extends GridManagerAdapter<SystemViewExporterSpi>
     implements ReadOnlySystemViewRegistry {
+    /** Class name for a SQL view exporter of system views. */
+    public static final String SYSTEM_VIEW_SQL_SPI = "org.apache.ignite.internal.managers.systemview.SqlViewExporterSpi";
```
Comment (agura) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java:61]: Visibility scope for this constant is too broad.

File: modules/indexing/src/main/java/org/apache/ignite/internal/managers/systemview/SqlViewExporterSpi.java [hunk 4]
```diff
@@ -36,12 +37,9 @@
...
  *
  * Note, instance of this class created with reflection.
- * @see IgnitionEx#SYSTEM_VIEW_SQL_SPI
+ * @see GridSystemViewManager#SYSTEM_VIEW_SQL_SPI
  */
-public class SqlViewExporterSpi extends IgniteSpiAdapter implements SystemViewExporterSpi {
-    /** System view filter. */
-    @Nullable private Predicate<SystemView<?>> filter;
-
+class SqlViewExporterSpi extends IgniteSpiAdapter implements SystemViewExporterSpi {
```
Comment (agura) [inline modules/indexing/src/main/java/org/apache/ignite/internal/managers/systemview/SqlViewExporterSpi.java:42]: IMHO, it isn't the best solution to have internal implementation of any SPI. May be we can extract some common parts of Exporter SPI into a new internal super-class which should be inherited by `SystemViewExporterSpi` and our internal analog of `SqlViewEXporterSpi`?

**Comment by agura** [comment]
Please, remove `:` after issue number in commit message. See the contribution checklist for details.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java (hunk 2 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java:91]: `SqlViewExporterSpi` now is package-private so it's forbidden for the user to create instances.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java (hunk 1 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java:85]: replaced.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java (hunk 3 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java:61]: fixed.

File: modules/indexing/src/main/java/org/apache/ignite/internal/managers/systemview/SqlViewExporterSpi.java (hunk 4 above)
Comment (nizhikov) [reply modules/indexing/src/main/java/org/apache/ignite/internal/managers/systemview/SqlViewExporterSpi.java]: `AbstractSystemViewExporterSpi` introduced. Please, take a look one more time.

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java (hunk 2 above)
Comment (agura) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java]: Any way it is just somewhat sloppy.

Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java]: OK, I've added a check to ensure `SqlViewExporterSpi` not presented in the initial array.

Comment (agura) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/GridSystemViewManager.java]: It's definitely redundant deu to a package privacy. I told about naming only. Please revert this check and merge.

**Pull request merged**